Surface_mesh/include/CGAL/Surface_mesh/Surface_mesh.h

```diff
@@ -495,6 +493,18 @@ class Surface_mesh
                while ( mesh_->has_valid_index(hnd_) && mesh_->is_removed(hnd_)) --hnd_;
         }
 
+        void advance(std::ptrdiff_t n)
+        {
+            CGAL_assertion(mesh_ != NULL);
+            CGAL_assertion(!(mesh_->has_garbage()));
+            hnd_ += n;
+        }
+
+        std::ptrdiff_t distance_to(const Index_iterator& other) const
+        {
+          return std::ptrdiff_t(other.hnd_) - std::ptrdiff_t(this->hnd_);
+        }
+      
```

Comment:
Houston, we have a problem!...

This commit makes a lot of BGL algorithms fail on `Surface_mesh`. @sloriot explained to me that this `distance_to` function is wrong were there is "garbage" in the sequence: elements that are marked as removed, but still there.

Reply:
All the full-red lines of the testsuite are because of this commit:
See for example:
https://cgal.geometryfactory.com/CGAL/testsuite/results-4.13-Ic-38.shtml#BGL